api/client/src/main/java/org/projectnessie/client/http/impl/jdk11/JavaHttpClient.java

```diff
@@ -75,6 +77,26 @@ public URI getBaseUri() {
 
   @Override
   public void close() {
+    if (client != null) {
+      client
+          .executor()
+          .ifPresent(
```

Comment:
I wonder: is the executor actually present?

Reply:
According to the javadocs, it's always present, a default one is created if none is provided.

Reply:
Then the javadoc's wrong :P

Reply:
Indeed it's confusing, the builder method states:

> Sets the executor to be used for asynchronous and dependent tasks. If this method is not invoked prior to building, a default executor is created for each newly built HttpClient.

And the getter method states:

> Even though this method may return an empty optional, the HttpClient may still have an non-exposed default executor that is used for executing asynchronous and dependent tasks.

🤷‍♂️

Reply:
I will check. 

@findepi: Could you please also share me the thread dump which proved the thread leak?